broker/submissions/routes.py

```diff
@@ -24,11 +25,18 @@ def generate_spreadsheet(submission_uuid):
 
     message = 'The spreadsheet is being generated.'
     if job_not_created(spreadsheet_job) or job_finished(spreadsheet_job):
-        spreadsheet_export_service = ExportToSpreadsheetService(app.ingest_api)
-        spreadsheet_export_service.async_export_and_save(submission_uuid, app.SPREADSHEET_STORAGE_DIR)
-        return response_json(HTTPStatus.ACCEPTED, {'message': message})
+        token = request.headers.get('Authorization')
+        if not token:
+            return response_json(HTTPStatus.BAD_REQUEST, {'status': HTTPStatus.BAD_REQUEST,
+                                                          'message': 'missing Authorization header'})
+        # need a dedicated ingest client that can be authenticated
+        ingest_api = IngestApi()
+        ingest_api.set_token(token)
+        spreadsheet_export_service = ExportToSpreadsheetService(app=app, ingest_api=ingest_api)
+        job_id = spreadsheet_export_service.async_export_and_save(submission_uuid, app.SPREADSHEET_STORAGE_DIR)
+        return response_json(HTTPStatus.ACCEPTED, {'message': message, 'job_id': job_id})
     else:
-        return response_json(HTTPStatus.ACCEPTED, {'message': message})
+        return response_json(HTTPStatus.ACCEPTED, {'message': message, 'job_id': spreadsheet_job.get('job_id')})
```

Comment:
I think that `ACCEPTED` in this case is an inaccurate status. I am not sure, though, what the correct one would be.